sleep_block.py

```diff
@@ -18,6 +21,48 @@ class Sleep(Block):
 
     interval = TimeDeltaProperty(title='Interval', default={'seconds': 1})
 
+    def __init__(self):
+        super().__init__()
+        self._signals = []
+        self._signals_lock = Lock()
+
+    def configure(self, context):
+        super().configure(context)
+        self._load_signals = self.persistence.load('signals') or []
+
+    def start(self):
+        super().start()
+        self._schedule_persistence_sleeps()
+
+    def stop(self):
+        super().stop()
+        self._store_signals(_time())
+        self.persistence.save()
+
     def process_signals(self, signals):
         # After interval, notify these signals
         emit = Job(self.notify_signals, self.interval, False, signals)
+        self._store_signals(_time() + self.interval.total_seconds(), signals)
+
+    def _schedule_persistence_sleeps(self):
+        """ Schedule emit jobs for signals loaded from persistence """
+        ctime = _time()
+        for (time, signals) in self._load_signals:
+            # get time to go before the signals should be notified
+            dtime = time - ctime
+            if dtime > 0 and signals:
+                dtime = timedelta(seconds=dtime)
+                Job(self.notify_signals, dtime, False, signals)
+                self._store_signals(time, signals)
```

Comment:
This is somewhat redundant to process_signals. Maybe add another method like `_emit_signals_after_duration` that stores the signals and sets up the job